yjit/src/asm/mod.rs

```diff
@@ -219,14 +225,14 @@ impl CodeBlock {
     }
 
     // Add a label reference at the current write position
-    pub fn label_ref(&mut self, label_idx: usize) {
+    pub fn label_ref<E: 'static>(&mut self, label_idx: usize, width: usize, encode: E) where E: FnOnce(&mut CodeBlock, i64, i64) {
         assert!(label_idx < self.label_addrs.len());
 
         // Keep track of the reference
-        self.label_refs.push(LabelRef {
-            pos: self.write_pos,
-            label_idx,
-        });
+        self.label_refs.push(LabelRef { pos: self.write_pos, label_idx, width, encode: Box::new(encode) });
+
+        // Move past however many bytes the instruction takes up
+        self.write_pos += width;
     }
```

Comment:
My very limited understanding is that since I'm accepting an `FnOnce` which is a trait, the object that implements it doesn't have a known size at compile time. But since `LabelRef` structs get put into a `Vec`, they have to have a size. So this becomes necessary. Though tbh there is _probably_ a better way, I just couldn't find it.

I thought about that other option but since the numbers could be close to boundaries where we then encode other instructions based on different offsets, I thought this might be simpler.

Reply:
This could get called multiple times though, may not want to make it FnOnce 🤔